src/python.gen.ts

```diff
@@ -376,7 +389,7 @@ ${this.hooks.join('\n')}
     }
 
     let attrsArgs = 'auto_attribs=True, kw_only=True'
-    if (type instanceof WriteType) {
+    if (this.inputTypes.has(type)) {
```

Comment:
I haven't grasped exactly what `inputTypes` does yet. Does the collection grow as as each method's parameters are added, and the constructor for a given method isn't required if the `inputType` has already been encountered?

Reply:
Oh .. is `inputTypes` so `__init__` is only generated for those models requiring it?

Reply:
yes, sorry if that wasn't clear - I can change the name to `methodInputModelTypes` or some such (and/or leave a code comment). I could have gone the route of manually generating constructors for all the models but I wanted to limit boilerplate/bloat as much as possible and at some point we can undo this when the code editors catch up with the `@attrs` auto-generated `__init__`s

Reply:
Thanks for confirming.